src/dist/block_service/block_service_manager.cpp

```diff
@@ -48,26 +49,25 @@ block_filesystem *block_service_manager::get_block_filesystem(const std::string
         if (iter != _fs_map.end())
             return iter->second.get();
 
-        const char *provider_type = dsn_config_get_value_string(
-            (std::string("block_service.") + provider).c_str(), "type", "", "block service type");
-
+        DSN_DEFINE_string((std::string("block_service.") + provider).c_str(),
+                          provider_type,
+                          "",
+                          "block service type");
         block_filesystem *fs =
-            utils::factory_store<block_filesystem>::create(provider_type, PROVIDER_TYPE_MAIN);
+            utils::factory_store<block_filesystem>::create(FLAGS_provider_type, PROVIDER_TYPE_MAIN);
         if (fs == nullptr) {
             derror("acquire block filesystem failed, provider = %s, provider_type = %s",
                    provider.c_str(),
-                   provider_type);
+                   FLAGS_provider_type);
             return nullptr;
         }
 
-        const char *arguments =
-            dsn_config_get_value_string((std::string("block_service.") + provider).c_str(),
-                                        "args",
-                                        "",
-                                        "args for block_service");
-
+        DSN_DEFINE_string((std::string("block_service.") + provider).c_str(),
+                          arguments,
```

Comment:
The configuration name here has changed

Reply:
flag api doesn't support such a variable definition of config. Do not refactor this place.